arcgis-ios-sdk-samples/Utility Network/Perform valve isolation trace/PerformValveIsolationTraceViewController.swift

```diff
@@ -0,0 +1,261 @@
+            if let error = error {
+                self.setStatus(message: "Loading starting location features failed.")
+                self.presentAlert(error: error)
+            } else if features == nil || features!.isEmpty {
+                self.setStatus(message: "Starting location features not found.")
+            } else {
+                // Get the geometry of the first feature for the starting location as a point.
+                guard let startingLocationGeometry = features!.first!.geometry as? AGSPoint else {
+                    self.setStatus(message: "Drawing starting location feature failed.")
+                    return
+                }
+                // Create a graphic for the starting point and add it to the graphics overlay.
+                let startingPointSymbol = AGSSimpleMarkerSymbol(style: .cross, color: .green, size: 25)
+                let startingLocationGraphic = AGSGraphic(geometry: startingLocationGeometry, symbol: startingPointSymbol)
+                let startingLocationGraphicsOverlay = AGSGraphicsOverlay()
+                startingLocationGraphicsOverlay.graphics.add(startingLocationGraphic)
+                self.mapView.graphicsOverlays.add(startingLocationGraphicsOverlay)
+                self.mapView.setViewpoint(AGSViewpoint(center: startingLocationGeometry, scale: 3000), completion: nil)
+            }
+        }
+    }
+    
+    func setStatus(message: String) {
+        statusLabel.text = message
+    }
+    
+    func setUIState() {
+        let utilityNetworkIsReady = utilityNetwork.loadStatus == .loaded
+        categoryButton.isEnabled = utilityNetworkIsReady
+        
+        let canTrace = utilityNetworkIsReady && startingLocationElement != nil
+        traceButton.isEnabled = canTrace
+    }
+    
+    /// Clear all the feature selections from previous trace.
+    func clearLayersSelection() {
+        self.mapView.map?.operationalLayers.lazy
+            .compactMap { $0 as? AGSFeatureLayer }
+            .forEach { $0.clearSelection() }
+    }
+    
+    // MARK: - Actions
+    
+    @IBAction func traceButtonTapped(_ button: UIBarButtonItem) {
+        guard let parameters = makeTraceParameters() else { return }
+        SVProgressHUD.show(withStatus: "Running isolation trace…")
+        utilityNetwork.trace(with: parameters) { [weak self] (traceResults, error) in
+            guard let self = self else { return }
+            if let error = error {
+                self.setStatus(message: "Trace failed.")
+                self.presentAlert(error: error)
+                SVProgressHUD.dismiss()
+                return
+            }
+            guard let elementTraceResult = traceResults?.first as? AGSUtilityElementTraceResult, !elementTraceResult.elements.isEmpty else {
+                self.setStatus(message: "Trace completed with no output.")
+                SVProgressHUD.dismiss()
+                return
+            }
+            self.clearLayersSelection()
```

Comment:
Should move this call to the top of the function - otherwise the previous graphics won't got cleared away when a error or invalid case happens.